test/kvao/delete.suite.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const bdd = require('../helpers/bdd-if');
+const helpers = require('./_helpers');
+const should = require('should');
+
+module.exports = function(dataSourceFactory, connectorCapabilities) {
+  describe('delete', function() {
+    let CacheItem;
+    beforeEach(function unpackContext() {
+      CacheItem = helpers.givenCacheItem(dataSourceFactory);
+      return CacheItem.deleteAll();
+    });
+
+    it('removes the keys-value pair associated to the given key',
+    function() {
+      return helpers.givenKeys(CacheItem, ['key1', 'key2'])
+        .then(() => CacheItem.delete('key1'))
+        .then(() => CacheItem.keys())
+        .then((keys) => {
+          should(keys).eql(['key2']);
+        });
+    });
+
+    it('does not remove key-value pairs unassociated to the given key',
```

Comment:
I wouldn't bother with this test. What kind of implementation would delete a single key across all models?